BetaCameras/Sick.VisionaryT.Pro/VisionaryTPro.cs

```diff
@@ -0,0 +1,398 @@
+                float yp2 = yp * yp;
+
+                for (int col = 0; col < width; col++)
+                {
+                    float xp = (intrinsics.Cx - col) / intrinsics.Fx;
+
+                    // correct the camera distortion
+                    float r2 = xp * xp + yp2;
+                    float r4 = r2 * r2;
+                    float k = 1 + intrinsics.K1 * r2 + intrinsics.K2 * r4;
+
+                    // Undistorted direction vector of the point
+                    float x = xp * k;
+                    float y = yp * k;
+                    float s0Inv = (float)(1 / Math.Sqrt(x * x + y * y + 1)); //z is 1, since x and y are coordinates on normalized image plane.
+
+                    directions[row, col] = new Point3f(-x * s0Inv, -y * s0Inv, s0Inv);
+                }
+            }
+            return directions;
+        }
+
+        #endregion
+
+        private void UpdateLoop()
+        {
+            int consecutiveFailCounter = 0;
+            while (!_cancelUpdateThreadSource.Token.IsCancellationRequested)
+            {
+                try
+                {
+                    // 0 - CameraModel
+                    // 1 - distance Image
+                    // 2 - intensity Image
+                    SyncCola();
+                    uint jsonSize = ReceiveDataSize();
+                    _backJsonData = ReceiveJsonString(jsonSize);
+                    if(null == _directions)
+                    {
+                        string json = new string(_backJsonData);
+                        var frameData = JsonConvert.DeserializeObject<List<CameraObject>>(json);                        
+                        _imageWidth = frameData[1].Data.Data.Width;
+                        _imageHeight = frameData[1].Data.Data.Height;
+                        InverseBrownConradyParams intrinsics = ParseIntrinsics(frameData[0]);
+                        _directions = CalcDirections(intrinsics, _imageWidth, _imageHeight);
+
+                        //Determine the offsets of intensity and distance image, which are stable over time.
+                        string needle = "\"data\":\"";
+                        int start_first_img = json.IndexOf(needle, 0) + needle.Length;
+                        int end_first_img = json.IndexOf("\"", start_first_img);
+                        _distanceJsonOffset = start_first_img;
+                        _distanceJsonSize = end_first_img - start_first_img;
+                        int start_second_img = json.IndexOf(needle, end_first_img) + needle.Length;
+                        int end_second_img = json.IndexOf("\"", start_second_img);
+                        _intensityJsonOffset = start_second_img;
+                        _intensityJsonSize = end_second_img - start_second_img;
+
+                        if ("uint16" != frameData[1].Data.Data.ImageType
+                        &&  "uint16" != frameData[2].Data.Data.ImageType)
```

Comment:
This check is actually broken, I think. IMHO it should be `||` instead of `&&`. Then the error message should output which `frameData` element has the wrong `ImageType`.

Same holds for the endianess check.